PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp

```diff
@@ -1949,23 +1949,83 @@ void ScintillaEditView::bufferUpdated(Buffer * buffer, int mask)
 	}
 }
 
+bool ScintillaEditView::isFoldIndentationBased() const
+{
+	const auto lexer = execute(SCI_GETLEXER);
+	// search IndentAmount in scintilla\lexers folder
+	return lexer == SCLEX_PYTHON
+		|| lexer == SCLEX_COFFEESCRIPT
+		|| lexer == SCLEX_HASKELL
+		|| lexer == SCLEX_NIMROD
+		|| lexer == SCLEX_VB
+		|| lexer == SCLEX_YAML
+	;
+}
+
 void ScintillaEditView::collapse(int level2Collapse, bool mode)
 {
 	execute(SCI_COLOURISE, 0, -1);
 
-	int maxLine = static_cast<int32_t>(execute(SCI_GETLINECOUNT));
+	const int maxLine = static_cast<int32_t>(execute(SCI_GETLINECOUNT));
+	int line = 0;
 
-	for (int line = 0; line < maxLine; ++line)
+	if (isFoldIndentationBased())

```

Comment:
In order to avoid the regression and for the sake of maintenance, the old code should be kept. 
Use `isFoldIndentationBased()` to call the new code.
So you can also simplify the new code.

Reply:
It's moved to a new method collapseFoldIndentationBased().
Don't know how that can be simplified.

Reply:
@zufuliu
It looks OK for me. Thank you.